modules/messages/src/lib.rs

```diff
@@ -305,6 +308,36 @@ pub mod pallet {
 				payload: encoded_payload,
 				fee: delivery_and_dispatch_fee,
 			});
+			// Guaranteed to be called outside only when the message is accepted.
+			// We assume that the maximum weight call back used is `single_message_callback_overhead`, so do not perform
```

Comment:
This is a bit suboptimal for cases where `OnMessageAccepted` callback is not set - the user needs to pay for more weight than necessary initially. Ideally we would have a separate `WeightInfo` method for that.

Reply:
Even if we have a separate `weightInfo` method, the user also needs to pre-paid callback weight, right?

Reply:
If it returns zero, then no :)

Reply:
(second option here https://github.com/paritytech/parity-bridges-common/pull/1134#issuecomment-918877463 )

Reply:
The second option is better indeed. But I have no idea about how to calculate the maximal accepted call back weight via benchmark in this case?  According to the block total weight?

Reply:
Yes, this option is a bit harder for runtime developer - you'll need to add benchmarks + you'll need tests to ensure that this `maximal_accepted_callback_weight` always returns weight that is >= than the weight from benchmark. But for the simplest case (where you only allow single db write + single db read per message) you may return `DbWeight::get().reads_writes(1, 1)` from the `maximal_accepted_callback_weight` without any benchmarks. Default implementation (`impl WeightInfoExt for () { ... }` should return zero.

That said, I'm ok with current implementation - both options have their pros and cons. But @tomusdrw has a valid concern here.

Reply:
@tomusdrw Please take some time to review again.